### PR TITLE
Switch to code routing API

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

**Comment by adamralph** [comment]
Thanks @SzymonPobiega!

Marking as WIP, since we will hold off merging until after NDC Sydney.

**Comment by SzymonPobiega** [comment]
Can this be merged now?

**Comment by adamralph** [comment]
@SzymonPobiega I'll review ASAP.

**Comment by adamralph** [comment]
@Particular/microservices-workshop-maintainers I'm struggling to find the time to review this, would anyone else be able to review it in the meantime?

**Comment by mauroservienti** [comment]
I can take this on Monday.

File: exercises/02-publish-subscribe/after/Divergent.Customers/Config/EndpointConfig.cs [hunk 1]
```diff
@@ -33,8 +34,13 @@ public void Customize(EndpointConfiguration endpointConfiguration)
...
             endpointConfiguration.Recoverability().Delayed(c=>c.NumberOfRetries(0));
             endpointConfiguration.UseContainer<AutofacBuilder>(c => c.ExistingLifetimeScope(container));
-            endpointConfiguration.UseTransport<MsmqTransport>()
-                .ConnectionString("deadLetter=false;journal=false");
+
+            var routing = endpointConfiguration.UseTransport<MsmqTransport>()
+                .ConnectionString("deadLetter=false;journal=false")
+                .Routing();
+
+            routing.RegisterPublisher(typeof(OrderSubmittedEvent), "Divergent.Sales");
```
Comment (dvdstelt) [inline exercises/02-publish-subscribe/after/Divergent.Customers/Config/EndpointConfig.cs]: How well did you test this everything? In XML config it included the entire assembly, now just a single message?

File: exercises/02-publish-subscribe/instructions.md [hunk 2]
```diff
@@ -95,17 +95,10 @@ The `OrderSubmittedHandler` should process the `OrderSubmittedEvent` published b
...
-```xml
-<configSections>
-  <section name="UnicastBusConfig" type="NServiceBus.Config.UnicastBusConfig, NServiceBus.Core" />
-</configSections>
-<UnicastBusConfig>
-  <MessageEndpointMappings>
-    <add Assembly="Divergent.Sales.Messages" Endpoint="Divergent.Sales" />
-  </MessageEndpointMappings>
-</UnicastBusConfig>
+In the `Divergent.Shipping` project, configure the publisher for the `OrderSubmittedEvent`. To do this use the `routing` object obtained when configuring the transport and add the following statement
```
Comment (dvdstelt) [inline exercises/02-publish-subscribe/instructions.md]: for the `OrderSubmittedEvent` event.

I sounds a bit odd, but then it should not include the word "the". Maybe we should remove the "Event" suffix and explain that events are in past tense? Or create another PR for that ;-)

**Comment by dvdstelt** [comment]
I haven't run through one or more of the exercises. How certain are you everything works? We can only be 100% sure if we go through each and every one of them! :(

**Comment by adamralph** [comment]
I believe, as part of the review, the reviewer should do a full run through of all the affected exercises, to ensure the attendee experience doesn't regress.

**Comment by dvdstelt** [comment]
@SzymonPobiega Could you please update the PR description and mention which issue this is related to? I'm a bit late with this, but I love creating digital paper trails so I can (later) understand where things are coming from.

@mauroservienti I can review this if you don't have time before your holiday.

**Comment by DavidBoike** [comment]
@dvdstelt Are you reviewing this?

**Comment by dvdstelt** [comment]
Yes

**Pull request merged**